src/FormatStream.js

```diff
@@ -44,10 +39,11 @@ class FormatStream extends stream.Transform {
    * @throws {TypeError} if not given a string as a topic
    */
   static _formatPayload (data, topic) {
+    topic = typeof topic == 'function' ? topic(data) : topic;
 
-    // Topic is required and must be a string!
-    if (!topic || typeof topic != 'string') {
-      throw new TypeError('formatData requires a topic, none given!')
+    if (typeof topic != 'string') {
```

Comment:
Not a big deal, I guess this is kinda of a hot topic in the JS community, but I usually use `=== ` and `!==` which just makes sure its the same type as well, I guess is kinda preference! :) I also use eslint to catch these "bugs" for me in my editor, so its easy to avoid :) Here is the content of one of my .eslintrc files: 
```
{
  "extends": "eslint:recommended",
  "parser": "babel-eslint",
  "env": {
    "es6": true,
    "browser": true,
    "jasmine": true,
    "node": true,
    "mocha": true
  },
  "ecmaFeatures": {
    "arrowFunctions": true,
    "blockBindings": true,
    "classes": true,
    "defaultParams": true,
    "destructuring": true,
    "forOf": true,
    "generators": true,
    "modules": true,
    "spread": true,
    "templateStrings": true,
    "jsx": true
  },
  "rules": {
    "max-len": [1, 120, 2, {ignoreComments: true}],
    "quote-props": [1, "consistent-as-needed"],
    "no-cond-assign": [2, "except-parens"],
    "radix": 0,
    "space-infix-ops": 0,
    "no-unused-vars": [1, {"vars": "local", "args": "none"}],
    "default-case": 0,
    "no-else-return": 0,
    "no-param-reassign": 0,
    "quotes": 0,
    "semi": [1, "always"],
    "eol-last": [1],
    "no-mixed-requires": [0],
    "no-underscore-dangle": [0],
    "no-undef": 1,
    "no-undef-init": 1,
    "no-undefined": 1
  }
}
```